releng_sop/pulp_clone_repos.py

```diff
@@ -0,0 +1,364 @@
+        self.pulp_config = PulpAdminConfig(self.env["pulp_server"])
+        self.skip_repo_check = skip_repo_check
+        self.content_category = content_category
+        self.pulp_password = self.pulp_config["client"].get("password")
+
+        self.data_from_all = {
+            "release_id": self.release_id,
+            "service": "pulp",
+            "repo_family": self.repo_family,
+            "content_format": "rpm",
+            "arch": self.arches,
+            "variant_uid": self.variants,
+            "content_category": self.content_category,
+            "shadow": False,
+        }
+
+        self.data_to_all = {
+            "release_id": self.release_id_to,
+            "service": "pulp",
+            "repo_family": self.repo_family,
+            "content_format": "rpm",
+            "arch": self.arches,
+            "variant_uid": self.variants,
+            "content_category": self.content_category,
+            "shadow": False,
+        }
+
+    def rearange(self, result):
+        """Creating dictionary from repos and to repos."""
+        self.rep = {}
+        for x in result:
+            key = (x['arch'], x['variant_uid'], x['content_category'])
+            if key in self.rep:
+                raise UsageError('Error same key in repos')
+            self.rep[key] = x['name']
+        return self.rep
+
+    def password_prompt(self, force=False, commit=False):
+        """Get password to authenticate with Pulp.
+
+        :param force:    Always ask for password, even if present in Pulp config
+        :type force:     Boolean
+
+        :param commit:   Flag to indicate if password is required for a commit action.
+                         If not set, password will not be asked for.
+        :type commit:    Boolean
+        """
+        if not commit:
+            return
+
+        result = self.pulp_password
+
+        if force:
+            result = ""
+
+        while not result:
+            msg = "Enter Pulp password for %s@%s: "
+            prompt = msg % (self.pulp_config["client"]["user"], self.pulp_config.name)
+            result = getpass.getpass(prompt=prompt)
+
+        self.pulp_password = result
+
+    def query_repo(self):
+        """Get name of pdc repo_from and pdc repo_to."""
+        if self.data_from_all['release_id'] == self.data_to_all['release_id']:
+            raise UsageError('Release id is same')
+        client = PDCClient(self.env["pdc_server"], develop=True)
+
+        self.result_from = client['content-delivery-repos']._(page_size=0, **self.data_from_all)
+        self.result_to = client['content-delivery-repos']._(page_size=0, **self.data_to_all)
+
+        if not self.result_from or not self.result_to:
+            self.result_from = []
+            self.result_to = []
+
+        if (len(self.result_from) != len(self.result_to)) and (not self.skip_repo_check):
+            raise UsageError('Error')
+
+        self.rep_from = {}
+        self.rep_to = {}
```

Comment:
Should `rep_from` and `rep_to` be used outside of this method? if not, these should be local variables instead of class members.
This empty initialization could be also removed, as the variables are initialized on the following lines.